diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3000af0..bc15297 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,7 @@
 cmake_minimum_required (VERSION 3.13)
 enable_language(Fortran)
 project (mcfm C CXX Fortran)
+include("GNUInstallDirs")
 
 if (CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
     set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -cpp -finit-local-zero -ffixed-line-length-none -ffree-line-length-512 -fopenmp -ffpe-summary=none")
@@ -99,6 +100,7 @@ option(lhapdf_include_path "LHAPDF include path" OFF)
 
 option(with_vvamp "Build with VVAmp for WW,WZ and ZZ processes" ON)
 
+include_directories(${CMAKE_BINARY_DIR}/local/include/qd ${CMAKE_BINARY_DIR}/local/include)
 string(ASCII 27 Esc)
 set(BoldRed "${Esc}[1;31m")
 set(ColorReset "${Esc}[m")
@@ -167,6 +169,7 @@ if(${with_library})
        target_include_directories(mltest2 PRIVATE src/Inc)
        target_compile_definitions(mltest2 PUBLIC LIB_SUFFIX="${CMAKE_SHARED_LIBRARY_SUFFIX}")
     endif()
+    install(TARGETS libmcfm DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 add_subdirectory(src)
@@ -334,10 +337,16 @@ set(CMAKE_INSTALL_RPATH "${CMAKE_BINARY_DIR}/local/lib")
 
 ## Installation
 
-install(FILES "${CMAKE_SOURCE_DIR}/Bin/process.DAT" DESTINATION "${CMAKE_BINARY_DIR}")
-install(FILES "${CMAKE_SOURCE_DIR}/Bin/br.sm1" DESTINATION "${CMAKE_BINARY_DIR}")
-install(FILES "${CMAKE_SOURCE_DIR}/Bin/br.sm2" DESTINATION "${CMAKE_BINARY_DIR}")
-install(DIRECTORY "${CMAKE_SOURCE_DIR}/Bin/PDFs" DESTINATION "${CMAKE_BINARY_DIR}")
+install(TARGETS mcfm DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(FILES "${CMAKE_SOURCE_DIR}/Bin/process.DAT" DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(FILES "${CMAKE_SOURCE_DIR}/Bin/br.sm1" DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(FILES "${CMAKE_SOURCE_DIR}/Bin/br.sm2" DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(FILES "${CMAKE_SOURCE_DIR}/src/Inc/MCFM/CXX_Interface.h" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/MCFM)
+install(FILES "${CMAKE_SOURCE_DIR}/src/Inc/MCFM/CXX_Wrapper.h" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/MCFM)
+install(FILES "${CMAKE_SOURCE_DIR}/src/Inc/MCFM/Flavor_Map.h" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/MCFM)
+if(use_internal_lhapdf)
+  install(DIRECTORY "${CMAKE_SOURCE_DIR}/Bin/PDFs" DESTINATION "${CMAKE_BINARY_DIR}")
+endif()
 
 if(${with_library})
     if ((DEFINED OLDIR) OR (DEFINED RCLDIR) OR (DEFINED MLDIR))
